qiskit/tools/qi/pauli.py

```diff
@@ -102,27 +102,49 @@ def to_matrix(self):
         Order is q_n x q_{n-1} .... q_0
         """
         x = np.array([[0, 1], [1, 0]], dtype=complex)
+        y = np.array([[0, -1j], [1j, 0]], dtype=complex)
         z = np.array([[1, 0], [0, -1]], dtype=complex)
         id_ = np.array([[1, 0], [0, 1]], dtype=complex)
-        x_temp = 1
+        matrix = 1
         for k in range(self.numberofqubits):
-            if self.v[k] == 0:
-                temp_z = id_
-            elif self.v[k] == 1:
-                temp_z = z
+            if self.v[k] == 0 and self.w[k] == 0:
+                new = id_
+            elif self.v[k] == 1 and self.w[k] == 0:
+                new = z
+            elif self.v[k] == 0 and self.w[k] == 1:
+                new = x
+            elif self.v[k] == 1 and self.w[k] == 1:
+                new = y
             else:
-                print('the z string is not of the form 0 and 1')
-            if self.w[k] == 0:
-                temp_x = id_
-            elif self.w[k] == 1:
-                temp_x = x
+                print('the string is not of the form 0 and 1')
+            matrix = np.kron(new, matrix)
+
+        return matrix
+
+    def to_spmatrix(self):
+        """Convert Pauli to a sparse matrix representation (CSR format).
+
+        Order is q_n x q_{n-1} .... q_0
+        """
```

Comment:
Can you add a note to this function stating what it returns? Something similar to (changing the `scipy.sparse.spmatrix` to the type of object it will return if needed):
```
"""Convert Pauli to a sparse matrix representation (CSR format).

Order is q_n x q_{n-1} .... q_0

Returns:
    scipy.sparse.spmatrix: a sparse matrix that ...
"""
```

Reply:
@diego-plan9 Thanks, I changed the codes to pass the lint check. By the way, I find that there is no return description for `to_label`, `to_matrix`. Do you want me to add them as well? Or just leave as is?

Reply:
Thanks - that would be awesome! Documenting `to_spmatrix()` would be enough, but if you can also take the change to document those other methods would be much appreciated.